src/ServiceLocatorInterface.php

```diff
@@ -15,9 +15,11 @@ interface ServiceLocatorInterface extends ContainerInterface
     /**
      * Build a service by its name, using optional options (such services are NEVER cached).
      *
-     * @param  string $name
+     * @template T of object
+     * @param  string|class-string<T> $name
      * @param  null|array<mixed>  $options
      * @return mixed
+     * @psalm-return ($name is class-string ? T : mixed)
```

Comment:
In theory, yes.
In practice, this is not necessarily true.

I think adding type inference is good though, even if it **may** be a lie :thinking: 

It's an accepted convention, so I'd say :+1:, and whoever breaks the convention is at fault here instead.

Reply:
Yeh, let's go with it 👍

Reply:
If we got it for `build`, that makes sense for `get` too, doesn't it?
`psr/container` would not accept it, but we can do it :wink: